### PR TITLE
feat: add cancel button

### PR DESCRIPTION
**Pull request by prachigarg19**
Show Cancel button in top panel
 Show Cancel button only if there is a callback or if user visits from donate.plant-for-the-planet.org

### CODE REVIEW

File: src/Common/CancelButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+const CancelButton = ({ returnUrl }: Props): ReactElement => {
+  const router = useRouter();
+
+  const cancelDonation = () => {
+    router.push(returnUrl);
+  };
+
+  return (
+    <button className={styles["cancel-button"]} onClick={cancelDonation}>
+      <CloseIcon />
```
Comment (mohitb35) [inline src/Common/CancelButton/index.tsx:19]: We're already using a `CloseIcon` in the app. Can we use the same icon?

File: src/Donations/LeftPanel/index.tsx [hunk 2]
```diff
@@ -26,10 +27,20 @@ function LeftPanel(): ReactElement {
...
 
-  const [isMobile, setIsMobile] = useState(false);
+  const router = useRouter();
+  const [showCancelButton, setShowCancelButton] = useState(false);
+
+  useEffect(() => {
+    if (!router.query.step) {
+      setShowCancelButton(true);
+    }
+  }, [router]);
```
Comment (mohitb35) [inline src/Donations/LeftPanel/index.tsx:38]: 1. Avoid confusion between the `showCancelButton` state variable, and the `canShowBackButton` / `canShowCancelButton` variable. Perhaps `showCancelButton` should be renamed to something that indicates what it represents better. 
2. Checking `step` as a proxy to decide which page the user came on won't work if the user comes directly to the step 1 page (the link usually has `to` and `step` is implicitly understood). Also, the cancel button will disappear if the user refreshes the page after clicking on a project from `Select Project`. Don't depend on state for this information.

**Comment by mariahosfeld** [comment]
When a user visits with a direct link (without callback provided), I would expect that we do not see the cancel button. Please check the behavior in this cases.

**Comment by mohitb35** [comment]
Closed as this approach was abandoned. Replaced by #420 which only changes the icon and makes a minor condition change.

**Pull request closed without merging**